src/main/java/org/sunbird/migrate/service/impl/UserMigrationServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+        String formattedCurrentTime = currentTime.format(formatter);
+
+        // Get yesterday's date at 1 AM in UTC
+        ZonedDateTime yesterdayAtOneAM = currentTime
+                .minusDays(1) // Subtract one day to get yesterday
+                .withHour(1)  // Set the hour to 1 AM
+                .withMinute(0)
+                .withSecond(0)
+                .withNano(0); // No need to set zone again as it's already in UTC
+
+        // Format yesterday's time
+        String formattedYesterdayAtOneAM = yesterdayAtOneAM.format(formatter);
+
+        // Construct the request body using Map
+        Map<String, Object> filters = new HashMap<>();
+        filters.put(Constants.PROFILE_DETAILS_PROFILE_STATUS, Constants.NOT_MY_USER);
+
+        // Create a separate HashMap for the inner filter
+        Map<String, String> innerFilter = new HashMap<>();
+        innerFilter.put("<=", formattedCurrentTime);
+        innerFilter.put(">=", formattedYesterdayAtOneAM);
+        filters.put(Constants.PROFILE_DETAILS_UPDATEDAS_NOT_MY_USER_ON, innerFilter);
+
+        Map<String, Object> request = new HashMap<>();
+        request.put(Constants.FILTERS, filters);
+
+        Map<String, Object> body = new HashMap<>();
+        body.put(Constants.REQUEST, request);
+
+        return body;
+    }
+
+    private Map<String, Object> getUserMigrateRequest(String userId, String channel, boolean isSelfMigrate) {
+        Map<String, Object> requestBody = new HashMap<String, Object>() {
+            {
+                put(Constants.USER_ID, userId);
+                put(Constants.CHANNEL, channel);
+                put(Constants.SOFT_DELETE_OLD_ORG, true);
+                put(Constants.NOTIFY_MIGRATION, false);
+                if (!isSelfMigrate) {
+                    put(Constants.FORCE_MIGRATION, true);
+                }
+            }
+        };
+        Map<String, Object> request = new HashMap<String, Object>() {
+            {
+                put(Constants.REQUEST, requestBody);
+            }
+        };
+        return request;
+    }
+
+    private String executeMigrateUser(Map<String, Object> request, Map<String, String> headers) {
+        String errMsg = StringUtils.EMPTY;
+        Map<String, Object> migrateResponse = (Map<String, Object>) outboundRequestHandlerService.fetchResultUsingPatch(
+                serverConfig.getSbUrl() + serverConfig.getLmsUserMigratePath(), request, headers);
+        if (migrateResponse == null
+                || !Constants.OK.equalsIgnoreCase((String) migrateResponse.get(Constants.RESPONSE_CODE))) {
+            errMsg = migrateResponse == null ? "Failed to migrate User."
+                    : (String) ((Map<String, Object>) migrateResponse.get(Constants.PARAMS))
+                    .get(Constants.ERROR_MESSAGE);
+        }
+        return errMsg;
+    }
+
+
+    private Map<String, Object> getUserExtPatchRequest(String userId, Map<String, Object> profileDetails, String defaultDepartment) {
```

Comment:
just create new Map object and add employmentDetails objects... this is should have only the parameters which you are intended to modify...